Contributing back jpgimenez changes to cbpi3

jpgimenez opened the pull request:
master branch includes the following changes:
  * check_hop_timer skip empty values https://github.com/Manuel83/craftbeerpi3/pull/236
  * PWM improvements https://github.com/Manuel83/craftbeerpi3/pull/237
  * migrate to python 3 https://github.com/Manuel83/craftbeerpi3/pull/257
  * add support for ssl https://github.com/Manuel83/craftbeerpi3/pull/249 (https://github.com/mstrinzha)
  * migrate to Flask 1.1 and python 3.8 https://github.com/Manuel83/craftbeerpi3/pull/259
  * Improve dummy sensor #260 

If you want to use this changes on your cbpi installation, download the patch https://patch-diff.githubusercontent.com/raw/Manuel83/craftbeerpi3/pull/258.patch and apply it.

Commit:
check_hop_timer skip empty values

Commit:
- GPIOPWM only set power if the actor is on
- create DummyPWM for PWM testing

Commit:
add space to propper logging the failing method name

mstrinzha committed:
Add support for SSL.

Remove hardcoded http protocol for polling.
bundle.js are using location.protocol instead of "http://" now.

Juan-gimenez-globant committed:
migrate to py3.5

Juan-gimenez-globant committed:
update ignores

Juan-gimenez-globant committed:
Merge branch 'python3-support'

dependabot[bot] committed:
Bump python-engineio from 0.9.2 to 3.8.2.post1

Bumps [python-engineio](https://github.com/miguelgrinberg/python-engineio) from 0.9.2 to 3.8.2.post1.
- [Release notes](https://github.com/miguelgrinberg/python-engineio/releases)
- [Changelog](https://github.com/miguelgrinberg/python-engineio/blob/master/CHANGES.md)
- [Commits](https://github.com/miguelgrinberg/python-engineio/commits)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump httplib2 from 0.9.2 to 0.18.0

Bumps [httplib2](https://github.com/httplib2/httplib2) from 0.9.2 to 0.18.0.
- [Release notes](https://github.com/httplib2/httplib2/releases)
- [Changelog](https://github.com/httplib2/httplib2/blob/master/CHANGELOG)
- [Commits](https://github.com/httplib2/httplib2/compare/0.9.2...v0.18.0)

Signed-off-by: dependabot[bot] <support@github.com>

jpgimenez committed:
Merge pull request #4 from jpgimenez/dependabot/pip/httplib2-0.18.0

Bump httplib2 from 0.9.2 to 0.18.0

dependabot[bot] committed:
Bump flask from 0.12.4 to 1.0

Bumps [flask](https://github.com/pallets/flask) from 0.12.4 to 1.0.
- [Release notes](https://github.com/pallets/flask/releases)
- [Changelog](https://github.com/pallets/flask/blob/master/CHANGES.rst)
- [Commits](https://github.com/pallets/flask/compare/0.12.4...1.0)

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump werkzeug from 0.11.10 to 0.15.3

Bumps [werkzeug](https://github.com/pallets/werkzeug) from 0.11.10 to 0.15.3.
- [Release notes](https://github.com/pallets/werkzeug/releases)
- [Changelog](https://github.com/pallets/werkzeug/blob/master/CHANGES.rst)
- [Commits](https://github.com/pallets/werkzeug/compare/0.11.10...0.15.3)

Signed-off-by: dependabot[bot] <support@github.com>

jpgimenez committed:
Merge pull request #3 from jpgimenez/dependabot/pip/flask-1.0

Bump flask from 0.12.4 to 1.0

jpgimenez committed:
Merge pull request #2 from jpgimenez/dependabot/pip/werkzeug-0.15.3

Bump werkzeug from 0.11.10 to 0.15.3

jpgimenez committed:
Merge pull request #1 from jpgimenez/dependabot/pip/python-engineio-3.8.2.post1

Bump python-engineio from 0.9.2 to 3.8.2.post1

jpgimenez committed:
updated plugins list

Juan-gimenez-globant committed:
migrate to Flask 1.1 and Python 3.8

Juan-gimenez-globant committed:
lint

Juan-gimenez-globant committed:
ignore .vscode

jpgimenez committed:
add eventlet dep...

jpgimenez committed:
python3 support

jpgimenez committed:
update python-engineio

jpgimenez committed:
Merge branch 'master' of github.com:jpgimenez/craftbeerpi3

jpgimenez committed:
use ImportError that is compatible with python 3.5

jpgimenez committed:
change update_addon to delete/download if the repo_url was changed

avollkopf commented:
Graphs are not diplayed with python 3. --> modules/log/endpoints.py requires change in line 59:
'''
        import csv

        array = []

        with open(filename, 'rt', encoding='utf-8') as f:

            reader = csv.reader(f)
'''

jpgimenez committed:
fix csv reading

jpgimenez committed:
lint endpoints

jpgimenez committed:
add settings to simulate a real sensor...

jpgimenez committed:
lint

jpgimenez committed:
add dev deps

jpgimenez committed:
Merge branch 'dummy-sensor-improvements'

jpgimenez committed:
fix charts

jpgimenez commented:
charts fixed df8c6a7

> Graphs are not diplayed with python 3. --> modules/log/endpoints.py requires change in line 59:
> '''
> import csv
> 
> ```
>     array = []
> 
>     with open(filename, 'rt', encoding='utf-8') as f:
> 
>         reader = csv.reader(f)
> ```
> 
> '''

lalo-uy commented:
I have added some code to display only one point per minute. 
Now loads ok a 5 days fermentation graph.

```
import csv
        array = []
        with open(filename, 'rb') as f:
            reader = csv.reader(f)
            l_time=0
            for row in reader:
                try:
                    p_time=int((datetime.datetime.strptime(row[0], "%Y-%m-%d %H:%M:%S") - datetime.datetime(1970, 1, 1)).total_seconds()) * 1000
                    if ( abs(float(row[1])) < 100 ) and ( p_time >= l_time ):
                        l_time = p_time + 60000;
                        array.append([p_time , float(row[1])])
                except:
                    pass
        return array
```

Also on lines 80 and 121  of    Modules/core/controler.py added tank name in the bottom of the graph.

  ```
   80         result.append({"name": kettle.name+" Temp", "data_type": "sensor", "data_id": kettle.sensor})

121              result.append({"name": fermenter.name+" Temp", "data_type": "sensor", "data_id": fermenter.senso
r})
```

avollkopf commented:
fermenter.py causes an error:
  File "/home/pi/craftbeerpi3/modules/fermenter/__init__.py", line 187, in start_fermentation
    inactive.direction = "C" if current_temp >= inactive.temp else "H"
TypeError: '>=' not supported between instances of 'float' and 'str'

-> fix: 
   inactive.direction = "C" if float(current_temp) >= float(inactive.temp) else "H"

jpgimenez committed:
fix up/down toogle

jpgimenez committed:
* migrate install to python3
* add Vagrantfile to test installation

jpgimenez commented:
> fermenter.py causes an error:
> File "/home/pi/craftbeerpi3/modules/fermenter/**init**.py", line 187, in start_fermentation
> inactive.direction = "C" if current_temp >= inactive.temp else "H"
> TypeError: '>=' not supported between instances of 'float' and 'str'
> 
> -> fix:
> inactive.direction = "C" if float(current_temp) >= float(inactive.temp) else "H"

I use cbpi only to brew, I ferment with brewpiless, but I'll try to fix this...

jpgimenez commented:
> I have added some code to display only one point per minute.
> Now loads ok a 5 days fermentation graph.
> 
> ```
> import csv
>         array = []
>         with open(filename, 'rb') as f:
>             reader = csv.reader(f)
>             l_time=0
>             for row in reader:
>                 try:
>                     p_time=int((datetime.datetime.strptime(row[0], "%Y-%m-%d %H:%M:%S") - datetime.datetime(1970, 1, 1)).total_seconds()) * 1000
>                     if ( abs(float(row[1])) < 100 ) and ( p_time >= l_time ):
>                         l_time = p_time + 60000;
>                         array.append([p_time , float(row[1])])
>                 except:
>                     pass
>         return array
> ```
> 
> Also on lines 80 and 121 of Modules/core/controler.py added tank name in the bottom of the graph.
> 
> ```
>  80         result.append({"name": kettle.name+" Temp", "data_type": "sensor", "data_id": kettle.sensor})
> 
> 121              result.append({"name": fermenter.name+" Temp", "data_type": "sensor", "data_id": fermenter.senso
> r})
> ```

could you create a PR against my branch? or maybe send me a patch...

jpgimenez committed:
typo

jpgimenez committed:
add dep

jpgimenez committed:
add pdbpp to dev req...

jpgimenez committed:
force temp to float

avollkopf commented:
> 
> I use cbpi only to brew, I ferment with brewpiless, but I'll try to fix this...

ok, i changed the code yesterday to the line i posted and the error was gone

lalo-uy commented:
Will investigate haw to create a PR.


El El dom, 9 de ago. de 2020 a la(s) 16:11, Juan Pablo Giménez <
notifications@github.com> escribió:

> I have added some code to display only one point per minute.
> Now loads ok a 5 days fermentation graph.
>
> import csv
>         array = []
>         with open(filename, 'rb') as f:
>             reader = csv.reader(f)
>             l_time=0
>             for row in reader:
>                 try:
>                     p_time=int((datetime.datetime.strptime(row[0], "%Y-%m-%d %H:%M:%S") - datetime.datetime(1970, 1, 1)).total_seconds()) * 1000
>                     if ( abs(float(row[1])) < 100 ) and ( p_time >= l_time ):
>                         l_time = p_time + 60000;
>                         array.append([p_time , float(row[1])])
>                 except:
>                     pass
>         return array
>
> Also on lines 80 and 121 of Modules/core/controler.py added tank name in
> the bottom of the graph.
>
>  80         result.append({"name": kettle.name+" Temp", "data_type": "sensor", "data_id": kettle.sensor})
>
> 121              result.append({"name": fermenter.name+" Temp", "data_type": "sensor", "data_id": fermenter.senso
> r})
>
> could you create a PR against my branch? or maybe send me a patch...
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/Manuel83/craftbeerpi3/pull/258#issuecomment-671089850>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACXBW4N7WDC5DOBFQVOSR5TR73YGBANCNFSM4PSSLBBQ>
> .
>

jpgimenez committed:
include PWM fix from https://github.com/PaulKGrimes/craftbeerpi3/tree/issue_242

jpgimenez committed:
fix for fermenter view

lalo-uy committed:
Update __init__.py

Remove reference to w1_bus_master1 since sensors could be no several bus masters like
# CraftBeerPi 1-wire support
dtoverlay=w1-gpio,gpiopin=4
dtoverlay=w1-gpio,gpiopin=27
dtoverlay=w1-gpio,gpiopin=18
:

jpgimenez committed:
Merge pull request #8 from lalo-uy/patch-3

Update __init__.py
